### PR TITLE
improve cancel/fill orders failure recovery

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_trading/personal_data/state.py [hunk 1]
```diff
@@ -179,6 +185,8 @@ async def synchronize_with_exchange(self, force_synchronization: bool = False) -
     async def refresh_operation(self):
         self.get_logger().debug("Starting refresh_operation")
         previous_state = self.state
+        # don't rely on self._underlying_refreshed_state only to avoid concurrency issues (use local variable instead)
```
[the remark this answers is not in the document]
Comment (GuillaumeDSM) [reply octobot_trading/personal_data/state.py:188]: right

File: octobot_trading/personal_data/orders/states/cancel_order_state.py [hunk 2]
```diff
@@ -68,10 +73,19 @@ async def _synchronize_with_exchange(self, force_synchronization: bool = False)
...
-        await super()._synchronize_with_exchange(force_synchronization=force_synchronization)
+        if self.is_max_synchronization_attempts_reached():
+            self.get_logger().info(
+                f"Forcing {enums.OrderStates.CANCELED.value} state after {self.MAX_SYNCHRONIZATION_ATTEMPTS} "
+                f"synchronization attempts for order {self.order}"
+            )
+            self._force_canceled_state()
+            asyncio.create_task(self.on_refresh_successful())
+        else:
+            await super()._synchronize_with_exchange(force_synchronization=force_synchronization)
```
[the remark this answers is not in the document]
Comment (GuillaumeDSM) [reply octobot_trading/personal_data/orders/states/cancel_order_state.py:88]: we can, I'll do it

**Comment by GuillaumeDSM** [comment]
up

**Review by Herklos — APPROVED** [review]
:+1:

**Pull request merged**